### DISCUSSION TITLE
How to mod and repack files inside .pack.zs

### DISCUSSION THREAD

**Discussion by Karlos007** [jordanbtucker/totk-no-loot-boxes#7] [Q&A] [answered]
Hi, that mod is awersome, I wanted to learn myself but I only edited easy unpacked bgyml files.

When I export this file and replace it (without even editing) 
![image](https://github.com/jordanbtucker/totk-no-loot-boxes/assets/23475289/92623dd9-df62-4ea6-afcd-88645fe8be24)
And then repacking to ResidentCommon.pack.zs the game just crashes

Is there any tutorial?

I used zstd-v1.5.5-win64 and Toolbox 1.0.202.0

**Comment by jordanbtucker** [comment, marked as answer]
I use [EX-King Editor](https://github.com/EXKing-Editor/EXKing-Editor) for the moment. See [DEVELOPMENT.md](https://github.com/jordanbtucker/totk-no-loot-boxes/blob/main/DEVELOPMENT.md) for how to pack this mod yourself and even make your own version of this mod.

**Comment by Karlos007** [comment]
Thank you!

**Comment by jordanbtucker** [comment]
You may also find useful this gist I made that lists the [vanilla drop tables](https://gist.github.com/jordanbtucker/a950e3a041b95867bc123b71c4cf7f28) for amiibo in TotK.

**Comment by Karlos007** [comment]
![image](https://github.com/jordanbtucker/totk-no-loot-boxes/assets/23475289/c89934e9-c9c5-41d0-8323-62974e032557)
much appreciated :)
finally got it to work

**Reply by jordanbtucker** [reply]
That's awesome! I literally just made a similar modification that made it drop 10 chests instead of one. I'm considering opening a poll to see if that's what people want.

**Reply by jordanbtucker** [reply]
I opened the poll.

https://github.com/jordanbtucker/totk-no-loot-boxes/discussions/9

**Comment by Karlos007** [comment]
Good idea 😀
I removed the food to avoid chests despawning

**Comment by Karlos007** [comment]
Gotta do the work again for v1.1.1

**Reply by jordanbtucker** [reply]
Did Nintendo make changes to `ResidentCommon.pack.zs`? If not, then there should be nothing to change.

**Comment by Karlos007** [comment]
oh yeah files are the same 
only 1.0.0 -> 1.1.0 changed ResidentCommon.pack.zs
It's gonna be a nice weekend with the ex king editor now that I know how it works
![image](https://github.com/jordanbtucker/totk-no-loot-boxes/assets/23475289/99220e42-3f50-4347-91dc-5d4f372535f2)
Too bad there is a spawn limit

**Reply by jordanbtucker** [reply]
I'm looking forward to what you come up with!

**Comment by Karlos007** [comment]
Nothing fancy lol 
But and idea, if you want to know what's inside the chest (Head/Upper/Lower) before opening
You can change the TBox_Field_Iron to TBox_Dungeon_Iron / TBox_Dungeon_Wood / TBox_Field_Wood
![image](https://github.com/jordanbtucker/totk-no-loot-boxes/assets/23475289/50be7a04-76fd-48b6-9dcf-19de6895710b)

(The stone chest gets stuck in the air, like spawning armor without chests)

**Reply by jordanbtucker** [reply]
That's genius!

**Reply by jordanbtucker** [reply]
Another thing that could be changed is to just not put weapons inside chests. Have them drop along side it.

**Comment by Karlos007** [comment]
Yeah that's also good, the only problem I found is that adding too many items makes the emulator crash on boot.
I'm still not sure if i will make a public mod.
Your version works perfect.
Maybe changing the chests model and weapons outside of chests.
